testgrid/cmd/entomologist/main.go

```diff
@@ -78,15 +78,29 @@ func (o *options) validate() error {
 	if o.repository == "" {
 		return errors.New("--github-repo is required")
 	}
-	if o.gcsPath == "" || !strings.HasPrefix(o.gcsPath, "gs://") {
-		return errors.New("invalid or missing --output")
+	if outputErr := validateOutput(o.output); outputErr != nil {
+		return outputErr
 	}
-	if o.gcsCredentials == "" {
+	if strings.HasPrefix(o.output, "gs://") && o.gcsCredentials == "" {
 		return errors.New("--gcs-credentials-file required for write operations")
 	}
 	return o.github.Validate(false)
 }
 
+func validateOutput(output string) error {
+	if output == "" {
+		return errors.New("missing --output")
+	}
+	if _, err := os.Stat(output); err != nil && !strings.HasPrefix(output, "gs://") {
```

Comment:
I think I would prefer it if the functionality that creates a new file was in main, and not hidden a few subfunctions deep in option-parsing. It makes it harder to know that this setup was done here.